homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +171,186 @@ def update_state(self, new_state):
+        service = SERVICE_VOLUME_DOWN if value else SERVICE_VOLUME_UP
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_input_source(self, value):
+        """Send input set value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set active_identifer to %s',
+                      self.entity_id, value)
+        source = self.sources[value]
+        self._flag[CHAR_ACTIVE_IDENTIFIER] = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_INPUT_SOURCE: source}
+        self.call_service(DOMAIN, SERVICE_SELECT_SOURCE, params)
+
+    def set_remote_key(self, value):
+        """Send remote key value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set remote key to %s', self.entity_id, value)
+        if value in MEDIA_PLAYER_KEYS:
+            service = MEDIA_PLAYER_KEYS[value]
+            # Handle Play Pause
+            if service == SERVICE_MEDIA_PLAY_PAUSE:
+                state = self.hass.states.get(self.entity_id).state
+                if state in (STATE_PLAYING, STATE_PAUSED):
+                    service = SERVICE_MEDIA_PLAY if state == STATE_PAUSED \
+                        else SERVICE_MEDIA_PAUSE
+            params = {ATTR_ENTITY_ID: self.entity_id}
+            self.call_service(DOMAIN, service, params)
+
+    def update_state(self, new_state):
+        """Update Television state after state changed."""
+        current_state = new_state.state
+
+        # Power state television
+        hk_state = current_state not in (STATE_OFF, STATE_UNKNOWN)
+        if not self._flag[CHAR_ACTIVE]:
+            hk_state = 1 if hk_state else 0
+            _LOGGER.debug('%s: Set current active state to %s',
+                          self.entity_id, hk_state)
+            self.char_active.set_value(hk_state)
+        self._flag[CHAR_ACTIVE] = False
+
+        # Set mute state
+        if CHAR_VOLUME_SELECTOR in self.chars_speaker:
+            current_mute_state = new_state.attributes.get(
+                ATTR_MEDIA_VOLUME_MUTED)
+            if not self._flag[FEATURE_TOGGLE_MUTE]:
+                _LOGGER.debug('%s: Set current mute state to %s',
+                              self.entity_id, current_mute_state)
+                self.char_mute.set_value(current_mute_state)
+            self._flag[FEATURE_TOGGLE_MUTE] = False
+
+        # Set active input
+        if self.support_select_source:
+            source_name = new_state.attributes.get(ATTR_INPUT_SOURCE)
+            if self.sources and not self._flag[CHAR_ACTIVE_IDENTIFIER]:
+                _LOGGER.debug('%s: Set current input to %s', self.entity_id,
+                              source_name)
+                if source_name in self.sources:
+                    index = self.sources.index(source_name)
+                    self.char_input_source.set_value(index)
+                else:
+                    self.char_input_source.set_value(0)
```

Comment:
During normal use this shouldn't happen. However if it does, it will create bug reports.
Can you add a `Warning` above to say that the source list got out of sync and the user should restart HA?